pkg/agent/route/route_windows.go

```diff
@@ -168,6 +176,89 @@ func (c *Client) DeleteRoutes(podCIDR *net.IPNet) error {
 	return nil
 }
 
+func (c *Client) AddServiceRoutes(svcIP net.IP, gwIP net.IP) error {
+	obj, found := c.hostRoutes.Load(svcIP.String())
+	svcIPNet := net.IPNet{IP: svcIP, Mask: net.CIDRMask(32, 32)}
+	metric := metricNormal
+	route := &netroute.Route{
+		LinkIndex:         c.nodeConfig.GatewayConfig.LinkIndex,
+		DestinationSubnet: &svcIPNet,
+		GatewayAddress:    gwIP,
+		RouteMetric:       int(metric),
+	}
+
+	if found {
+		existingRoute := obj.(*netroute.Route)
+		if existingRoute.GatewayAddress.Equal(route.GatewayAddress) && existingRoute.RouteMetric == route.RouteMetric {
+			klog.V(4).Infof("Route for Service IP `%s` already exists via gateway `%s` with metric `%d`",
+				svcIP.String(), gwIP.String(), metric)
+			return nil
+		}
+		// Remove the existing route if gateway or metric is not as expected.
+		if err := c.nr.RemoveNetRoute(existingRoute.LinkIndex, existingRoute.DestinationSubnet, existingRoute.GatewayAddress); err != nil {
+			klog.Errorf("Failed to delete existing route entry with destination `%s` gateway `%s`",
+				existingRoute.DestinationSubnet.String(), existingRoute.GatewayAddress.String())
```

Comment:
please use structured logging. I see that you did it in other places in this PR.

Reply:
Updated.